apps/arena/lib/arena/game_updater.ex

```diff
@@ -1848,6 +1867,16 @@ defmodule Arena.GameUpdater do
     game_state
   end
 
+  defp remove_pool_effects_from_entities(game_state, pool, entities) do
+    Enum.reduce(entities, game_state, fn {entity_id, _entity}, acc ->
+      if entity_id in pool.collides_with and pool.aditional_info.status == :READY do
+        Effect.remove_owner_effects(acc, entity_id, pool.id)
+      else
+        acc
+      end
+    end)
+  end
```

Comment:
If you're changing the game state only for a pool that is :READY, we should use pattern-match to only modify the game state in that case and just return the state otherwise.
Here, we are traversing the entities map just to return the same game state.

Reply:
This was more complicated than it should, i reworked it